jina/types/arrays/search_ops.py

```diff
@@ -112,3 +113,21 @@ def shuffle(self, seed: Optional[int] = None) -> 'DocumentArray':
         from .document import DocumentArray
 
         return DocumentArray(self.sample(len(self), seed=seed))
+
+    def split(self, attribute: str) -> Dict[Any, 'DocumentArray']:
+        """Split the :class:`DocumentArray` into multiple :class:`DocumentArray` according to the attribute value of
+        each :class:`Document`.
+
+        :param attribute: the attribute name to split stored in tags.
+        :return: a dict where Documents with the same value on `attribute` are grouped together, their orders are
+        preserved from the original :class:`DocumentArray`.
+        """
+        from .document import DocumentArray
+
+        rv = defaultdict(DocumentArray)
+        for doc in self:
+            value = doc.tags.get(attribute)
+            if not value:
+                continue
```

Comment:
We need to add a warning message when the `attribute` does not exist in the `Document`. At least this skipping behavior should be claimed in the docstring.
```suggestion
                continue
```

Reply:
please only label this behavior in docstring